model_analyzer/config/generate/optuna_run_config_generator.py

```diff
@@ -137,21 +143,47 @@ def get_configs(self) -> Generator[RunConfig, None, None]:
         RunConfig
             The next RunConfig generated by this class
         """
+        logger.info(
+            "Measuring default configuration to establish a baseline measurement"
+        )
         default_run_config = self._create_default_run_config()
         yield default_run_config
-        self._default_measurement = self._last_measurement
 
-        max_configs_to_search = self._determine_maximum_number_of_configs_to_search()
+        self._capture_default_measurement(default_run_config)
+        self._set_best_measurement(default_run_config)
+        self._print_debug_search_space_info()
 
+        max_configs_to_search = self._determine_maximum_number_of_configs_to_search()
         # TODO: TMA-1885: Need an early exit strategy
-        for _ in range(max_configs_to_search):
+        for trial_count in range(max_configs_to_search):
             trial = self._study.ask()
             trial_objectives = self._create_trial_objectives(trial)
+            logger.debug(f"Trial {trial_count+1} of {max_configs_to_search}:")
             run_config = self._create_objective_based_run_config(trial_objectives)
             yield run_config
+
             score = self._calculate_score()
+            self._set_best_measurement(run_config, score)
+            self._print_debug_score_info(run_config, score)
+
             self._study.tell(trial, score)
 
+    def _capture_default_measurement(self, default_run_config: RunConfig) -> None:
+        if not self._last_measurement:
+            raise TritonModelAnalyzerException(
+                "Default configuration did not return a measurement. Please check PA/Tritonserver log files."
+            )
+
+        self._default_measurement = self._last_measurement
+
+    def _set_best_measurement(self, run_config: RunConfig, score: float = 0) -> None:
+        if self._best_config_score is None:
+            self._best_config_name = run_config.model_variants_name()
+            self._best_config_score = 0
```

Comment:
why is this 0? Shouldn't the score still be whatever was passed in?

Reply:
That's correct. I've updated the code to reflect that score can always be used.